Additional features

NoyFlat opened the pull request:
Add option to delete comments

NoyFlat committed:
Add an option to request number of comments to display
Add timestamp field to Comment.java
Edit the display comments according to time submitted
Add form to html to select number of comments
Edit url in fetch to match the number of commnets chosen

NoyFlat committed:
Add option to delete comments
Add servlet DeleteDataServlet
Add delete button to each comment
Add comment class in css
Edit script.js - moved part of function getCommentsFromServer to different function
Add to Comment.java id and timestamp fields for sorting and deleting of comments.

NoyFlat committed:
Add option to delete comments
Add servlet DeleteDataServlet
Add delete button to each comment
Add comment class in css
Edit script.js - moved part of function getCommentsFromServer to different function

NoyFlat committed:
Update style.css

orarch commented on portfolio/src/main/java/com/google/sps/data/Comment.java:
```diff
@@ -19,12 +19,19 @@
+  private final long id;
 
-  public Comment(String name, String likedGame, String comment){
+  public Comment(long id, String name, String likedGame, String content, long timestamp){
```
I suggest reading about Builder design pattern in Java:

https://dzone.com/articles/design-patterns-the-builder-pattern

It's optional and might be out of scope for the portfolio project, but it's good to know for future reference :)

orarch commented on portfolio/src/main/webapp/index.html:
```diff
@@ -4,7 +4,7 @@
     <title>My Portfolio</title>
     <link rel="stylesheet" href="style.css">
-    <script src="script.js"></script>
+    <script src="script2.js"></script>
```
?

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -35,19 +37,23 @@
     
     List<Comment> comments = new ArrayList<>();
     // Iterate over all comments in datastore and add them to array
     for (Entity entity : results.asIterable()) {
```
This is kind of wasteful, you're fetching all the rows.
The limiting should be at the lowest level possible (when the query is being executed). 

Hint: use the PreparedQuery#asIterable method with the FetchOptions argument.

orarch reviewed (commented):
Overall nice PR. 

However, I don't think it fits the "small PR" best practice.
It's not well contained, it consists two features: 1. delete 2. limiting comments

NoyFlat committed:
Change according to Or's notes
Add builder to Comment
Change the loop in DataServlet doGet to have a limit insted of checking all comments
Chang html script to correct one

NoyFlat committed:
Merge branch 'additional-features' of https://github.com/NoyFlat/my-portfolio into additional-features

NoyFlat committed:
Fix accidental duplicate code on Comment.java
